include/deal.II/base/data_out_base.h

```diff
@@ -348,6 +348,149 @@ namespace DataOutBase
+   * for zero-dimensional patches into @p static variables that exist only
+   * once in the entire program, as opposed to once per patch. Specifically,
+   * this is the case for the @p neighbors array and the @p n_subdivisions
+   * member variable that make no sense for zero-dimensional patches because
+   * points have no natural neighbors across their non-existent faces, nor
+   * can they reasonably be subdivided.
+   *
+   * @author Wolfgang Bangerth, 2017.
+   */
+  template <int spacedim>
+  struct Patch<0,spacedim>
+  {
+    /**
+     * Make the <tt>spacedim</tt> template parameter available.
+     */
+    static const unsigned int space_dim=spacedim;
+
+    /**
+     * Corner points of a patch.  For the current class of zero-dimensional
+     * patches, there is of course only a single vertex.
+     *
+     * If <code>points_are_available==true</code>, then
+     * the coordinates of the point at which output is to be generated
+     * is attached as an additional row to the <code>data</code> table.
+     */
+    Point<spacedim> vertices[1];
+
+    /**
+     * An unused, @p static variable that exists only to allow access
+     * from general code in a generic fashion.
+     */
+    static unsigned int neighbors[1];
+
+    /**
+     * Number of this patch. Since we are not sure patches are handled in the
+     * same order, always, we better store this.
+     */
+    unsigned int patch_index;
+
+    /**
+     * Number of subdivisions with which this patch is to be written.
+     * <tt>1</tt> means no subdivision, <tt>2</tt> means bisection, <tt>3</tt>
+     * trisection, etc.
+     *
+     * Since subdivision makes no sense for zero-dimensional patches,
+     * this variable is no t used but exists only to allow access
+     * from general code in a generic fashion.
+     */
+    static unsigned int n_subdivisions;
+
+    /**
+     * Data vectors. The format is as follows: <tt>data(i,.)</tt> denotes the
+     * data belonging to the <tt>i</tt>th data vector. <tt>data.n_cols()</tt>
+     * therefore equals the number of output points; this number is
+     * of course one for the current class, given that we produce output on
+     * points. <tt>data.n_rows()</tt> equals the number of
+     * data vectors. For the current purpose, a data vector equals one scalar,
+     * even if multiple scalars may later be interpreted as vectors.
+     *
+     * Within each column, <tt>data(.,j)</tt> are the data values at the
+     * output point <tt>j</tt>; for the current class, @p j can only
+     * be zero.
+     *
+     * Since the number of data vectors is usually the same for all patches to
+     * be printed, <tt>data.size()</tt> should yield the same value for all
+     * patches provided. The exception are patches for which
+     * points_are_available are set, where the actual coordinates of the point
+     * are appended to the 'data' field, see the documentation of the
+     * points_are_available flag.
```

Comment:
I suggest we strip the general documentation and simply state the behavior in the special case.

Reply:
(This also applies to a couple of other places.)

Reply:
I read through this again, but I'd like to keep it there because it explains the purpose and reason of why we chose this data structure. For this particular specialization, the only valid index is zero, so one could argue that instead of storing a 2d table, a 1d vector would be good enough. The documentation explains why that's not what we do.